Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -600,19 +613,25 @@ const PipelineProgram* ProgramShaderCache::GetPipelineProgram(const OGLShader* v
   // Set program variables on the shader which will be returned.
   // This is only needed for drivers which don't support binding layout.
   prog->shader.SetProgramVariables();
-  auto ip = pipelineprograms.emplace(key, std::move(prog));
+
+  // If this is a shared context, ensure we sync before we return the program to
+  // the main thread. If we don't do this, some driver can lock up (e.g. AMD).
```

Comment:
Does this have some impact on performance / stuttering? If so, should it be in DriverDetails?

Reply:
My concern was that glFinish() would sync the driver thread. However, I've tested on NVIDIA and it doesn't seem to make any difference, but only by eye as there was too much time variance on a run-to-run basis to be conclusive. @JMC47 also couldn't see any difference.

Without any clear disadvantage, I think it's probably safest to leave it here, as resource sharing between contexts is likely not a highly tested path for most drivers. And if these races do exist, avoids potentially difficult-to-debug crashes.